From 5c8d61bc7eea44427ae47f03cb5353417853a4eb Mon Sep 17 00:00:00 2001
From: David Sernelius <david.sernelius@flir.se>
Date: Fri, 1 Nov 2019 13:13:31 +0100
Subject: [PATCH] ov5640: Update Lens correction and night mode settings.

Updated Lens correction and night mode settings provided by Anders Wistrand.
---
 .../media/platform/mxc/capture/ov5640_v2.c    | 125 +++++++++---------
 1 file changed, 65 insertions(+), 60 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/ov5640_v2.c b/drivers/media/platform/mxc/capture/ov5640_v2.c
index 41b6d96a51dd..f9bad2c0e803 100644
--- a/drivers/media/platform/mxc/capture/ov5640_v2.c
+++ b/drivers/media/platform/mxc/capture/ov5640_v2.c
@@ -260,7 +260,7 @@ static struct reg_value ov5640_init_setting_30fps_BT656[] = {
 	{0x3c0a, 0x9c, 0x0, 0x0},
 	{0x3c0b, 0x40, 0x0, 0x0},
 #endif
-	{0x3a00, 0x58, 0x0, 0x0},
+	{0x3a00, 0x5c, 0x0, 0x0}, //night mode 191223
 	{0x4001, 0x02, 0x0, 0x0},
 	{0x4004, 0x02, 0x0, 0x0},
 	{0x4005, 0x1a, 0x0, 0x0},
@@ -394,68 +394,69 @@ static struct reg_value ov5640_init_setting_30fps_BT656[] = {
 	{0x558b, 0xf8, 0x0, 0x0},
 
 	/* Lens correction */
-	{0x5800, 0x23, 0x0, 0x0},
-	{0x5801, 0x14, 0x0, 0x0},
-	{0x5802, 0x0f, 0x0, 0x0},
-	{0x5803, 0x0f, 0x0, 0x0},
-	{0x5804, 0x12, 0x0, 0x0},
-	{0x5805, 0x26, 0x0, 0x0},
-	{0x5806, 0x0c, 0x0, 0x0},
-	{0x5807, 0x08, 0x0, 0x0},
-	{0x5808, 0x05, 0x0, 0x0},
-	{0x5809, 0x05, 0x0, 0x0},
-	{0x580a, 0x08, 0x0, 0x0},
-	{0x580b, 0x0d, 0x0, 0x0},
-	{0x580c, 0x08, 0x0, 0x0},
-	{0x580d, 0x03, 0x0, 0x0},
+	/*LENC start 191023*/
+	{0x5800, 0x3f, 0x0, 0x0},
+	{0x5801, 0x3f, 0x0, 0x0},
+	{0x5802, 0x2c, 0x0, 0x0},
+	{0x5803, 0x2c, 0x0, 0x0},
+	{0x5804, 0x3d, 0x0, 0x0},
+	{0x5805, 0x3f, 0x0, 0x0},
+	{0x5806, 0x3f, 0x0, 0x0},
+	{0x5807, 0x1d, 0x0, 0x0},
+	{0x5808, 0x0d, 0x0, 0x0},
+	{0x5809, 0x0d, 0x0, 0x0},
+	{0x580a, 0x1c, 0x0, 0x0},
+	{0x580b, 0x3e, 0x0, 0x0},
+	{0x580c, 0x2c, 0x0, 0x0},
+	{0x580d, 0x0b, 0x0, 0x0},
 	{0x580e, 0x00, 0x0, 0x0},
 	{0x580f, 0x00, 0x0, 0x0},
-	{0x5810, 0x03, 0x0, 0x0},
-	{0x5811, 0x09, 0x0, 0x0},
-	{0x5812, 0x07, 0x0, 0x0},
-	{0x5813, 0x03, 0x0, 0x0},
+	{0x5810, 0x0a, 0x0, 0x0},
+	{0x5811, 0x2c, 0x0, 0x0},
+	{0x5812, 0x2b, 0x0, 0x0},
+	{0x5813, 0x0b, 0x0, 0x0},
 	{0x5814, 0x00, 0x0, 0x0},
-	{0x5815, 0x01, 0x0, 0x0},
-	{0x5816, 0x03, 0x0, 0x0},
-	{0x5817, 0x08, 0x0, 0x0},
-	{0x5818, 0x0d, 0x0, 0x0},
-	{0x5819, 0x08, 0x0, 0x0},
-	{0x581a, 0x05, 0x0, 0x0},
-	{0x581b, 0x06, 0x0, 0x0},
-	{0x581c, 0x08, 0x0, 0x0},
-	{0x581d, 0x0e, 0x0, 0x0},
-	{0x581e, 0x29, 0x0, 0x0},
-	{0x581f, 0x17, 0x0, 0x0},
-	{0x5820, 0x11, 0x0, 0x0},
-	{0x5821, 0x11, 0x0, 0x0},
-	{0x5822, 0x15, 0x0, 0x0},
-	{0x5823, 0x28, 0x0, 0x0},
-	{0x5824, 0x46, 0x0, 0x0},
-	{0x5825, 0x26, 0x0, 0x0},
-	{0x5826, 0x08, 0x0, 0x0},
-	{0x5827, 0x26, 0x0, 0x0},
-	{0x5828, 0x64, 0x0, 0x0},
-	{0x5829, 0x26, 0x0, 0x0},
-	{0x582a, 0x24, 0x0, 0x0},
-	{0x582b, 0x22, 0x0, 0x0},
-	{0x582c, 0x24, 0x0, 0x0},
-	{0x582d, 0x24, 0x0, 0x0},
-	{0x582e, 0x06, 0x0, 0x0},
-	{0x582f, 0x22, 0x0, 0x0},
-	{0x5830, 0x40, 0x0, 0x0},
-	{0x5831, 0x42, 0x0, 0x0},
-	{0x5832, 0x24, 0x0, 0x0},
-	{0x5833, 0x26, 0x0, 0x0},
-	{0x5834, 0x24, 0x0, 0x0},
-	{0x5835, 0x22, 0x0, 0x0},
-	{0x5836, 0x22, 0x0, 0x0},
-	{0x5837, 0x26, 0x0, 0x0},
-	{0x5838, 0x44, 0x0, 0x0},
-	{0x5839, 0x24, 0x0, 0x0},
-	{0x583a, 0x26, 0x0, 0x0},
-	{0x583b, 0x28, 0x0, 0x0},
-	{0x583c, 0x42, 0x0, 0x0},
-	{0x583d, 0xce, 0x0, 0x0},
+	{0x5815, 0x00, 0x0, 0x0},
+	{0x5816, 0x0b, 0x0, 0x0},
+	{0x5817, 0x2e, 0x0, 0x0},
+	{0x5818, 0x3a, 0x0, 0x0},
+	{0x5819, 0x1b, 0x0, 0x0},
+	{0x581a, 0x0c, 0x0, 0x0},
+	{0x581b, 0x0d, 0x0, 0x0},
+	{0x581c, 0x1e, 0x0, 0x0},
+	{0x581d, 0x3f, 0x0, 0x0},
+	{0x581e, 0x3f, 0x0, 0x0},
+	{0x581f, 0x3f, 0x0, 0x0},
+	{0x5820, 0x31, 0x0, 0x0},
+	{0x5821, 0x34, 0x0, 0x0},
+	{0x5822, 0x3f, 0x0, 0x0},
+	{0x5823, 0x3f, 0x0, 0x0},
+	{0x5824, 0x4d, 0x0, 0x0},
+	{0x5825, 0x1f, 0x0, 0x0},
+	{0x5826, 0x0f, 0x0, 0x0},
+	{0x5827, 0x1f, 0x0, 0x0},
+	{0x5828, 0x49, 0x0, 0x0},
+	{0x5829, 0x1f, 0x0, 0x0},
+	{0x582a, 0x3a, 0x0, 0x0},
+	{0x582b, 0x47, 0x0, 0x0},
+	{0x582c, 0x39, 0x0, 0x0},
+	{0x582d, 0x1b, 0x0, 0x0},
+	{0x582e, 0x1d, 0x0, 0x0},
+	{0x582f, 0x44, 0x0, 0x0},
+	{0x5830, 0x50, 0x0, 0x0},
+	{0x5831, 0x44, 0x0, 0x0},
+	{0x5832, 0x2a, 0x0, 0x0},
+	{0x5833, 0x1f, 0x0, 0x0},
+	{0x5834, 0x2a, 0x0, 0x0},
+	{0x5835, 0x37, 0x0, 0x0},
+	{0x5836, 0x28, 0x0, 0x0},
+	{0x5837, 0x2a, 0x0, 0x0},
+	{0x5838, 0x2b, 0x0, 0x0},
+	{0x5839, 0x1d, 0x0, 0x0},
+	{0x583a, 0x1d, 0x0, 0x0},
+	{0x583b, 0x1e, 0x0, 0x0},
+	{0x583c, 0x4c, 0x0, 0x0},
+	{0x583d, 0xcd, 0x0, 0x0},
 	/* end */
 
 	/* configure resolution */
@@ -519,6 +520,10 @@ static struct reg_value ov5640_init_setting_30fps_BT656[] = {
 	{0x3a14, 0x03, 0x0, 0x0},
 	{0x3a15, 0xd8, 0x0, 0x0},
 #endif
+	{0x3a02, 0x06, 0x0, 0x0}, //60Hz min frame rate [15:8] 9Hz 191023
+	{0x3a03, 0x50, 0x0, 0x0}, //60Hz min frame rate [7:0] 9Hz 191023
+	{0x3a14, 0x06, 0x0, 0x0}, //50Hz min frame rate [15:8] 9Hz 191023
+	{0x3a15, 0x50, 0x0, 0x0}, //50Hz min frame rate [7:0] 9Hz 191023
 	/* format config */
 	{0x4719, 0x01, 0x0, 0x0},
 	{0x4730, 0x01, 0x0, 0x0},	//CCIR656
-- 
2.17.1

